### ISSUE TITLE
vacation at end of TOC can cause flooding

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#59]
In testing my new TOC code, I set the speed up to 200 and got into a situation where the game was spinning heads-up (i.e., there was not the customary 30 second delay between toggling which of the two remaining players is on vacation).

Right before it happened the status showed the wrong amount of chips total in play, which I think is s a separate bug, but I'm reporting it here because I haven't yet been able to save the errant output due to my web client getting overwhelmed by all the output that is thrown at it due to this flooding bug.

Although I don't have an explanation for the missing chips, the flooding is due to the way all-ins on the ante interact with vacation:
```
<mb2:#table-1> Dealing No-Limit Texas Hold’em
<mb2:#table-1> deadhead antes 5100 as the big blind and is all-in
<mb2:#table-1> dh2 blinds 9800 and is all-in
<mb2:#table-1> deadhead is all in and does not post a blind
<mb2:#table-1> 1     deadhead  0 5100     0  [  ] [  ]
<mb2:#table-1> 2 B >V dh2      0       9800  [  ] [  ]
*mb2*               Your cards are: [2h] [2s]
<mb2:#table-1> 1     deadhead  0 5100     0  [  ] [  ]
<mb2:#table-1> 2 B >V dh2      0       9800  [  ] [  ]
<mb2:#table-1> Board: 6c 5c 7h
*mb2*               Your cards are: [2h] [2s]
<mb2:#table-1> 1     deadhead  0 5100     0  [  ] [  ]
<mb2:#table-1> 2 B >V dh2      0       9800  [  ] [  ]
<mb2:#table-1> Board: 6c 5c 7h Ks
*mb2*               Your cards are: [2h] [2s]
<mb2:#table-1> 1     deadhead  0 5100     0  [  ] [  ]
<mb2:#table-1> 2 B >V dh2      0       9800  [  ] [  ]
<mb2:#table-1> Board: 6c 5c 7h Ks 4c
*mb2*               Your cards are: [2h] [2s]
<mb2:#table-1> Uncalled 9800 returned to dh2
<mb2:#table-1> deadhead  Does Not Show
<mb2:#table-1> deadhead wins 5100
<mb2:#table-1> Dealing No-Limit Texas Hold’em
<mb2:#table-1> dh2 antes 9800 as the big blind and is all-in
<mb2:#table-1> deadhead blinds 5100 and is all-in
<mb2:#table-1> dh2 is all in and does not post a blind
<mb2:#table-1> 1 B   deadhead  0       5100  [  ] [  ]
<mb2:#table-1> 2   >V dh2      0 9800     0  [  ] [  ]
*mb2*               Your cards are: [Ac] [9d]
<mb2:#table-1> 1 B   deadhead  0       5100  [  ] [  ]
<mb2:#table-1> 2   >V dh2      0 9800     0  [  ] [  ]
<mb2:#table-1> Board: Ad 9h 2h
*mb2*               Your cards are: [Ac] [9d]
<mb2:#table-1> 1 B   deadhead  0       5100  [  ] [  ]
<mb2:#table-1> 2   >V dh2      0 9800     0  [  ] [  ]
<mb2:#table-1> Board: Ad 9h 2h Kh
*mb2*               Your cards are: [Ac] [9d]
<mb2:#table-1> 1 B   deadhead  0       5100  [  ] [  ]
<mb2:#table-1> 2   >V dh2      0 9800     0  [  ] [  ]
<mb2:#table-1> Board: Ad 9h 2h Kh Qs
*mb2*               Your cards are: [Ac] [9d]
<mb2:#table-1> Uncalled 5100 returned to deadhead
<mb2:#table-1> dh2  Does Not Show
<mb2:#table-1> dh2 wins 9800
<mb2:#table-1> Dealing No-Limit Texas Hold’em
<mb2:#table-1> deadhead antes 5100 as the big blind and is all-in
<mb2:#table-1> dh2 blinds 9800 and is all-in
<mb2:#table-1> deadhead is all in and does not post a blind
<mb2:#table-1> 1     deadhead  0 5100     0  [  ] [  ]
<mb2:#table-1> 2 B >V dh2      0       9800  [  ] [  ]
*mb2*               Your cards are: [8d] [Kd]
<mb2:#table-1> 1     deadhead  0 5100     0  [  ] [  ]
<mb2:#table-1> 2 B >V dh2      0       9800  [  ] [  ]
<mb2:#table-1> Board: 7s Qd Th
*mb2*               Your cards are: [8d] [Kd]
<mb2:#table-1> 1     deadhead  0 5100     0  [  ] [  ]
<mb2:#table-1> 2 B >V dh2      0       9800  [  ] [  ]
<mb2:#table-1> Board: 7s Qd Th Td
*mb2*               Your cards are: [8d] [Kd]
<mb2:#table-1> 1     deadhead  0 5100     0  [  ] [  ]
<mb2:#table-1> 2 B >V dh2      0       9800  [  ] [  ]
<mb2:#table-1> Board: 7s Qd Th Td 4h
*mb2*               Your cards are: [8d] [Kd]
<mb2:#table-1> Uncalled 9800 returned to dh2
<mb2:#table-1> deadhead  Does Not Show
<mb2:#table-1> deadhead wins 5100
```
FWIW, the total on the table is stable at 5,100 + 9,800 = 14,900, which is 5,100 less than should be on the table.

**Comment by ctm** [comment]
I rushed this bug report and hadn't had a chance to look carefully at what is happening, but I've done so now. The issue is that the wrong player is awarded the pot.

Once the antes and blinds get big enough, anyone on vacation should lose very quickly as that players chips get awarded to the player who isn't on vacation. mb2 already has code to make sure that at least one player is not on vacation, specifically to prevent this sort of flooding. However, for some reason the pot is being awarded to the player on vacation as though the other player isn't in the pot. My guess is this bug and #60 are both regressions and are related. Once I fix them, the flooding issue will automatically go away.

**Comment by ctm** [comment]
Now that I've fixed #60, I've verified that the flooding no longer happens.
